### PR TITLE
[FL-3388] NFC/RFID detector 

### PR DESCRIPTION
**Pull request by Skorpionm**
# What's new

- [FL-3388] NFC/RFID detector 
- Furi_Hal_Rfid: added field validation functions to furi_hal_rfid
- Furi_Hal_Nfc: added field validation functions to furi_hal_nfc

# Verification 

- Launch Applications -> Tools->NFC/RFID detector. bring to the NFC, RFID readers and check the field definitions by the application

# Checklist (For Reviewer)

- [x] PR has description of feature/bug or link to Confluence/Jira task
- [x] Description contains actions to verify feature/bugfix
- [x] I've built this code, uploaded it to the device and verified feature/bugfix

[FL-3388]: https://flipperzero.atlassian.net/browse/FL-3388?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: applications/main/subghz/subghz_cli.c [hunk 1]
```diff
@@ -196,6 +196,50 @@ void subghz_cli_command_tx(Cli* cli, FuriString* args, void* context) {
...
 
+#include "fild_validation_rfid_driver.h"
+#include <furi_hal_nfc.h>
+#include <lib/ST25RFAL002/include/rfal_nfc.h>
+#include <lib/ST25RFAL002/source/st25r3916/st25r3916_com.h>
+void subghz_cli_command_us(Cli* cli, FuriString* args, void* context) {
+    UNUSED(context);
+    UNUSED(args);
+
+    printf("Check frecuency. Press CTRL+C to stop\r\n");
```
Comment (wosk) [inline applications/main/subghz/subghz_cli.c]:
```suggestion
    printf("Check frequency. Press CTRL+C to stop\r\n");
```
and other occurrences

Comment (Skorpionm) [reply applications/main/subghz/subghz_cli.c]: this is a test build, it won't be in the main, but thanks anyway

File: firmware/targets/f7/furi_hal/furi_hal_nfc.h [hunk 2]
```diff
@@ -423,6 +423,10 @@ FuriHalNfcReturn furi_hal_nfc_ll_txrx_bits(
 
 void furi_hal_nfc_ll_poll();
 
+void furi_hal_nfc_check_vield_start();
+
+bool furi_hal_nfc_check_is_vield();
+
```
Comment (gornekich) [inline firmware/targets/f7/furi_hal/furi_hal_nfc.h]: field :)

Comment (Skorpionm) [reply firmware/targets/f7/furi_hal/furi_hal_nfc.h]: +

**Review by skotopes — APPROVED** [review]

**Pull request merged**